functional-tests/utils.py

```diff
@@ -464,3 +465,30 @@ def setup_test_logger(datadir_root: str, test_name: str) -> logging.Logger:
     logger.addHandler(stream_handler)
 
     return logger
+
+
+def get_envelope_pushdata(inp: str):
+    op_if = "63"
+    op_endif = "68"
+    op_pushbytes_33 = "21"
+    op_false = "00"
+    start_position = inp.index(f"{op_false}{op_if}")
+    end_position = inp.index(f"{op_endif}{op_pushbytes_33}", start_position)
+    op_if_block = inp[start_position + 3 : end_position]
+    op_pushdata = "4d"
+    pushdata_position = op_if_block.index(f"{op_pushdata}")
+    # we don't want PUSHDATA + num bytes b401
+    return op_if_block[pushdata_position + 2 + 4 :]
+
+
+def submit_da_blob(btcrpc: BitcoindClient, seqrpc: JsonrpcClient, blobdata: str):
+    _ = seqrpc.strataadmin_submitDABlob(blobdata)
+
+    # Allow some time for sequencer to publish blob
+    time.sleep(SEQ_PUBLISH_BATCH_INTERVAL_SECS)
```

Comment:
Can we replace this for `wait_until`?
Only if we cannot or does not make sense to use `wait_until` we should use `time.sleep`.

`wait_until` is more efficient. Especially if you customize the kwargs.

Reply:
submitDABlob doesn't provide txid so there is no direct way other than comparing previous last_l1_txid status. I guess i should just do that. Also i think there is this edge case where our new checkpoint batch might be posted after posting submitted DA blob, which is also be handled.